src/main/java/duke/tasktypes/Deadline.java

```diff
@@ -0,0 +1,67 @@
+package duke.tasktypes;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.format.FormatStyle;
+
+import duke.exceptions.DukeException;
+
+/**
+ * Represents a Deadline Task.
+ * Deadline Tasks are unique in having an ending time.
+ */
+public class Deadline extends Task {
+
+    private LocalDate byDate;
+    private LocalTime byTime;
+    private LocalDateTime byDateTime;
+    private String forSaving;
+    private String doneBy;
+
+    /**
+     * Constructs a Deadline Task.
+     * Deadline Tasks are constructed with an ending time.
+     *
+     * @param description Description of task.
+     * @param by Ending time of task.
+     */
+    public Deadline(String description, String by) throws DukeException {
+        super(description);
+        String[] dateAndTime = by.split(" ");
+        String date = dateAndTime[0];
+        String time = dateAndTime[1];
+        date = date.replace('/', '-');
+        this.forSaving = date + " " + time;
+        time = time.substring(0, 2) + ':' + time.substring(2);
```

Comment:
Interesting way of getting manipulating the string to get your time. Would it be better if you have a DatetimeParser class that would handle the string input and return you time ?